apps/arweave/src/ar_webhook.erl

```diff
@@ -239,16 +300,306 @@ do_call_webhook(URL, Headers, Entity, Event, _N) ->
 	]),
 	ok.
 
-entity_id(#block { indep_hash = ID }) -> ar_util:encode(ID);
-entity_id(#tx { id = ID }) -> ar_util:encode(ID).
+entity_id(#block{ indep_hash = ID }) -> ar_util:encode(ID);
+entity_id(#tx{ id = ID }) -> ar_util:encode(ID);
+entity_id(#{ txid := TXID }) -> TXID.
 
-to_json(#block {} = Block) ->
+to_json(#block{} = Block) ->
 	{JSONKVPairs} = ar_serialize:block_to_json_struct(Block),
 	JSONStruct = {lists:keydelete(wallet_list, 1, JSONKVPairs)},
 	ar_serialize:jsonify({[{block, JSONStruct}]});
-to_json(#tx {} = TX) ->
+to_json(#tx{} = TX) ->
 	{JSONKVPairs1} = ar_serialize:tx_to_json_struct(TX),
 	JSONKVPairs2 = lists:keydelete(data, 1, JSONKVPairs1),
 	JSONKVPairs3 = [{data_size, TX#tx.data_size} | JSONKVPairs2],
 	JSONStruct = {JSONKVPairs3},
-	ar_serialize:jsonify({[{transaction, JSONStruct}]}).
+	ar_serialize:jsonify({[{transaction, JSONStruct}]});
+to_json(Map) when is_map(Map) ->
+	jiffy:encode(Map).
+
+handle_sync_record_add_range(Start, End, StoreID, State) ->
+	handle_sync_record_add_range(Start, End, StoreID, State, 0).
+
+handle_sync_record_add_range(Start, End, StoreID, State, N) when N < ?NUMBER_OF_TRIES ->
+	case get_tx_offset_data(Start, End, State#state.tx_offset_cache) of
+		{ok, Data, Cache} ->
+			process_updated_tx_data(Data, StoreID, State#state{ tx_offset_cache = Cache });
+		not_found ->
+			State;
+		Error ->
+			?LOG_WARNING([{event, failed_to_process_webhook_sync_record_add_range},
+					{error, io_lib:format("~p", [Error])},
+					{range_start, Start},
+					{range_end, End}]),
+			timer:sleep(?WAIT_BETWEEN_TRIES),
+			handle_sync_record_add_range(Start, End, State, N + 1)
+	end;
+handle_sync_record_add_range(Start, End, _StoreID, State, _N) ->
+	?LOG_WARNING([{event, gave_up_webhook_tx_offset_data_fetch},
+		{range_start, Start},
+		{range_end, End},
+		{number_of_tries, ?NUMBER_OF_TRIES},
+		{wait_between_tries, ?WAIT_BETWEEN_TRIES}
+	]),
+	State.
+
+process_updated_tx_data([], _StoreID, State) ->
+	State;
+process_updated_tx_data([{TXID, Start, End} | Data], StoreID, State) ->
+	Cache = State#state.tx_offset_cache,
+	case cache_is_tx_data_marked_synced(TXID, Cache) of
+		true ->
+			process_updated_tx_data(Data, StoreID, State);
+		false ->
+			process_updated_tx_data(Data, StoreID,
+					maybe_call_transaction_data_synced(Start, End, TXID, StoreID, State))
```

Comment:
```suggestion
					maybe_call_transaction_data_synced_webhook(Start, End, TXID, StoreID, State))
```

(I don't feel strongly about this rename - I think it's clearer, but also makes a long function longer....)